packages/shared/src/session/waitForAuthInfoWithTimeout.ts

```diff
@@ -0,0 +1,11 @@
+import { raceWithTimeout } from "../async/raceWithTimeout";
+import { WAIT_FOR_AUTH_INFO_TIMEOUT } from "./config";
+import { deferredAuthInfo } from "./deferred";
+
+const promise = raceWithTimeout(deferredAuthInfo.promise, WAIT_FOR_AUTH_INFO_TIMEOUT);
```

Comment:
This starts the race at the module initialization time and not when `initializeSession` gets called. This is probably fine in practice today but I feel like it's a little bit less "correct"

Reply:
After a second thought... maybe that's intended so we don't wait too long before flushing logger queues. I'm not sure though

Reply:
I considered this and decided this was okay/intended (and also probably not too impactful since we'll likely call the wait-for method very near the start of the session anyway) but I could be convinced to change it.

I also considered if each caller should get their own start time or if there should be a global timeout and I decided the latter was preferable.